### ISSUE TITLE
Failed to stop the muxer

### ISSUE THREAD

**Issue by Vitor238** [deepmedia/Transcoder#95]
I am trying to cut a video recorded by my phone. When cutting, the following error is displayed:
`Failed to stop the muxer`
video path : ` /storage/emulated/0/Android/data/com.vitor.videoeditor/files/Movies/Speeacht/VID_20200718_121633.mp4`
destine path:  `/storage/emulated/0/Android/data/com.vitor.videoeditor/files/Movies/Speeacht/VID_20200718_121643.mp4`

my code:

```
 private fun clipVideos() {

        videoPath = intent.extras?.getString("video")
        Log.i("VIDEOEDITOR","path - $videoPath")
        val uri = Uri.fromFile(File(videoPath))

        val dataSource = UriDataSource(this,uri)
        val duration = dataSource.durationUs

        val startTime: Long = 0
        val endTime: Long = if (duration > 3000) 3000 else duration
        val clip = ClipDataSource(dataSource, startTime, endTime)

        val diretorioSaida = FileUtils.getVideosDir(this).absolutePath

        val destPath = (diretorioSaida + File.separator + FileUtils.VIDEO_PREFIX
                + LocalTimeUtils.getFormatedTime()
                + FileUtils.VIDEO_EXTENSION)

        Log.i("VIDEOEDITOR","Dest - $destPath")

        Transcoder.into(destPath)
                .addDataSource(clip)
                .setListener(object : TranscoderListener {
                    override fun onTranscodeCompleted(successCode: Int) {
                        videoList.add(destPath)
                        recyclerVideos.adapter?.notifyDataSetChanged()
                        toast("OK!")
                    }

                    override fun onTranscodeProgress(progress: Double) {
                        Log.i("PROGRESS", progress.toString())
                    }

                    override fun onTranscodeCanceled() {
                        toast("CANCELED")
                    }

                    override fun onTranscodeFailed(exception: Throwable) {
                        Log.i("VIDEOEDITOR",exception.message.toString())
                    }

                }).transcode()


    }

```

How to fix this error?
Another question: is it possible to cut a video for example every 3 seconds?

**Comment by pankaj89** [comment]
In my case when i speed up or down the video i am facing same issue
```
W/DefaultDataSink: Failed to release the muxer.
    java.lang.IllegalStateException: Failed to stop the muxer
        at android.media.MediaMuxer.nativeStop(Native Method)
        at android.media.MediaMuxer.stop(MediaMuxer.java:466)
        at android.media.MediaMuxer.release(MediaMuxer.java:706)

```
and 
```
java.lang.IllegalStateException: writeSampleData returned an error
at android.media.MediaMuxer.nativeWriteSampleData(Native Method)
at android.media.MediaMuxer.writeSampleData(MediaMuxer.java:694)
at com.otaliastudios.transcoder.sink.DefaultDataSink.writeTrack(DefaultDataSink.java:151)
at com.otaliastudios.transcoder.transcode.BaseTrackTranscoder.drainEncoder(BaseTrackTranscoder.java:288)
at com.otaliastudios.transcoder.transcode.BaseTrackTranscoder.transcode(BaseTrackTranscoder.java:163)
at com.otaliastudios.transcoder.engine.Engine.transcode(Engine.java:373)
at com.otaliastudios.transcoder.Transcoder$1.call(Transcoder.java:150)
at com.otaliastudios.transcoder.Transcoder$1.call(Transcoder.java:140)
```

Please provide some solution.

**Comment by usmanrana-ilsa** [comment]
Facing same issue.

**Comment by jumperson** [comment]
May be fix this issue by #159.

**Comment by natario1** [comment]
(Pasted to all my answers today: it's been a long time since my last issues review. I am sorry about the delay and I know you likely have moved on by now. Still, I'm going to answer where I can)

Closing as a "duplicate" of #139
